### ISSUE TITLE
Documentation about `class-const` needs to be updated.

### ISSUE THREAD

**Issue by duwensi-bgi** [pylint-dev/pylint#9724]
### Bug description

```python
"""
Saved as file a.py.
"""

import typing
from typing import ClassVar, Final


class Test:
    """Test class."""

    WITH_CLASS_VAR_1: ClassVar = 1
    WITH_CLASS_VAR_2: typing.ClassVar = 1
    WITH_FINAL_1: Final = 1
    WITH_FINAL_2: typing.Final = 1
```

### Configuration

```ini
# Saved as file pylintrc.
[BASIC]
class-attribute-naming-style=snake_case
class-const-naming-style=UPPER_CASE
```

### Command used

```shell
pylint --rcfile pylintrc a.py
```

### Pylint output

```python
a.py:12:4: C0103: Class attribute name "WITH_CLASS_VAR_1" doesn't conform to snake_case naming style (invalid-name)
a.py:13:4: C0103: Class attribute name "WITH_CLASS_VAR_2" doesn't conform to snake_case naming style (invalid-name)
```

### Expected behavior

According to this merged PR [#4279](https://github.com/pylint-dev/pylint/pull/4279)  and the pratise above, documentation about `class-const` in the table of setion *Additional details* for [v2](https://pylint.readthedocs.io/en/v2.15.10/user_guide/messages/convention/invalid-name.html?highlight=class-const-naming-style) and [v3](https://pylint.readthedocs.io/en/stable/user_guide/messages/convention/invalid-name.html) pylint should be:

`class-const` : Enum constants and class variables annotated with `Final`.

Currently is:

`class-const` : Enum constants and class variables annotated with `ClassVar`.

### Pylint version

```shell
astroid 2.12.13
pylint 2.15.9
Python 3.10.4 (tags/v3.10.4:9d38120, Mar 23 2022, 23:13:41) [MSC v.1929 64 bit (AMD64)] on win32
```

### OS / Environment

Windows 10
